### PR TITLE
rawgpsd: publish ephemerides

### PR DESCRIPTION
**Pull request by haraschax**

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
@gast04 can you test this in the GPS box with the Quectel device?

File: selfdrive/locationd/laikad.py [hunk 1]
```diff
@@ -129,9 +130,27 @@ def read_report(self, gnss_msg):
...
+      return gnss_msg.which == 'drSvPoly'
+    else:
+      return gnss_msg.which == 'ephemeris'
+
+  def read_ephemeris(self, gnss_msg):
+    # TODO this only works on GLONASS
+    if self.use_qcom:
+      # TODO this is not robust to gps week rollover
+      if self.gps_week is not None:
+        ephem = parse_qcom_ephem(gnss_msg.ephemeris, self.gps_week)
```
Comment (gast04) [inline selfdrive/locationd/laikad.py:144]: this does not work, `gnss_msg` in qcom has the structure `drSvPoly` (see is_ephemeris) and this one does not contain an ephemeris field

Comment (gast04) [inline selfdrive/locationd/laikad.py:144]: `gnss_msg.ephemeris` -> `gnss_msg.drSvPoly`

File: selfdrive/locationd/laikad.py [hunk 2]
```diff
@@ -348,6 +367,17 @@ def main(sm=None, pm=None):
 
     if sm.updated[raw_gnss_socket]:
       gnss_msg = sm[raw_gnss_socket]
+
+      # prefilter gnss messages
+      if gnss_msg.which() in ["measurementReport", "drMeasurementReport"]:
+        if getattr(gnss_msg, gnss_msg.which()).source not in ['glonass', 'gps', 'beidou', 'sbas']:
+          continue
+
+        if getattr(gnss_msg, gnss_msg.which()).gpsWeek > 32768:
```
Comment (adeebshihadeh) [inline selfdrive/locationd/laikad.py:376]: use numpy.iinfo instead of hardcoding this size

File: selfdrive/locationd/laikad.py [hunk 3]
```diff
@@ -129,9 +130,28 @@ def read_report(self, gnss_msg):
       new_meas = read_raw_ublox(report)
     return week, tow, new_meas
 
+  def is_ephemeris(self, gnss_msg):
+    if self.use_qcom:
+      return gnss_msg.which == 'drSvPoly'
+    else:
+      return gnss_msg.which == 'ephemeris'
```
Comment (adeebshihadeh) [inline selfdrive/locationd/laikad.py:137]: does this work and not just always return false? which is a function

**Pull request merged**